packages/matchbox/src/components/Modal/Modal.js

```diff
@@ -1,88 +1,144 @@
+`;
 
-    /**
-     * An optional function that is called on key down 'Escape' and on click outside modal content
-     */
-    onClose: PropTypes.func,
+const StyledCloseButton = styled(Button)`
+  ${closeButton}
+`;
 
-    /**
-     * Modal content
-     */
-    children: PropTypes.node,
+const StyledFocusLock = styled(FocusLock)`
+  ${focusLock}
+`;
 
-    showCloseButton: PropTypes.bool
+const StyledContent = styled('div')`
+  ${content}
+  ${contentAnimation}
+`;
+
+function Modal(props) {
+  const { onClose, children, containerId, className, showCloseButton, ...rest } = props;
+  let container = useRef(null);
+  let content = useRef(null);
+
+  const handleKeydown = e => {
+    const { open } = props;
+
+    if (open && onClose) {
+      onKey('escape', onClose)(e);
+    }
   };
 
-  handleOutsideClick = (e) => {
-    const { open, onClose } = this.props;
-    const isOutside = this.content && !this.content.contains(e.target) && this.container && this.container.contains(e.target);
+  const handleOutsideClick = e => {
+    const { open } = props;
+    const isOutside =
+      content && !content.contains(e.target) && container && container.contains(e.target);
 
     if (open && isOutside && onClose) {
       onClose(e);
     }
-  }
+  };
 
-  handleKeyDown = (e) => {
-    const { onClose, open } = this.props;
+  return (
+    <Portal containerId={containerId}>
+      <StyledBase
+        {...rest}
+        className={className}
+        onClose={onClose}
+        ref={el => (container = el)}
+        role="dialog"
+        aria-modal="true"
+      >
+        <StyledWrapper>
+          <ModalContent open={props.open} maxWidth={props.maxWidth}>
+            <div ref={el => (content = el)}>
+              <WindowEvent event="keydown" handler={handleKeydown} />
+
+              <WindowEvent event="click" handler={handleOutsideClick} />
 
-    if (open && onClose) {
-      onKey('escape', onClose)(e);
-    }
-  }
-
-  render() {
-    const {
-      onClose,
-      open,
-      children,
-      className,
-      showCloseButton,
-      ...rest
-    } = this.props;
-
-    const modalClasses = classnames(
-      styles.Modal,
-      open && styles.open,
-      className
-    );
-
-    return (
-      <div className={modalClasses} onClose={onClose} {...rest} ref={(node) => this.container = node} role="dialog" aria-modal="true">
-        <Grid center='xs' middle='xs' className={styles.Grid}>
-          <Grid.Column xs={11} md={9} xl={7}>
-            <Content contentRef={(node) => this.content = node} open={open}>
-              <WindowEvent event='keydown' handler={this.handleKeyDown} />
-              <WindowEvent event='click' handler={this.handleOutsideClick} />
               {showCloseButton && (
-                <Button className={styles.CloseButton} flat onClick={onClose} data-id='modal-close'>
-                  Close <Close />
-                </Button>
+                <StyledCloseButton flat onClick={onClose} data-id="modal-close">
```

Comment:
:/ Need to add `cursor: pointer` to the Button component – have a ticket for that

Reply:
Yeeeah I saw that ticket so decided not to add it here.